pkg/forward/forward.go

```diff
@@ -260,8 +260,13 @@ func (isdf *InterStepDataForward) forwardAChunk(ctx context.Context) {
 			isdf.opts.logger.Errorw("failed to applyUDF", zap.Error(err))
 			return
 		}
-		// update toBuffers
+
 		for _, message := range m.writeMessages {
+			// Update isLate to reflect the message event time change after being processed by source data transformer.
+			if isdf.opts.vertexType == dfv1.VertexTypeSource && processorWM.After(message.EventTime) {
```

Comment:
this has a problem, `processorWM` could make every data to be late if transformer does a decr operation on eventTime.